File: #1971.find-if-path-exists-in-graph.cpp
class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {

        vector<vector<int>> v(n);
        for(vector<int> e : edges){
            v[e[0]].push_back(e[1]);
            v[e[1]].push_back(e[0]);
        }

        stack<int> s;
        s.push(source);
        vector<int> v2(n,0);
        while (!s.empty()) {
            int node = s.top();
            s.pop();
            if (node == destination) {
                return true;
            }

            if (v2[node] == 1) {
                continue;
            }

            v2[node] = 1;
            for (int i: v[node]) {
                s.push(i);
            }
        }
        return false;
    }
};

File: #1971.find-if-path-exists-in-graph.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @param {number} source
 * @param {number} destination
 * @return {boolean}
 */
var validPath = function(n, edges, source, destination) {
    const arr1 = Array.from(Array(n), () => new Array());
    for (const [V, W] of edges) {
        arr1[V].push(W);
        arr1[W].push(V);
    }

    const arr2 = new Array(n).fill(false);
    const validPath = (node) => {
        arr2[node] = true;
        for (const neighbor of arr1[node]) {
            if (!arr2[neighbor]) {
                validPath(neighbor);
            }
        }
    };

    validPath(source);
    return arr2[destination];
};
